service/constants/constants.js

```diff
@@ -37,4 +48,32 @@ const Constants = {
     POST_PURPOSE_CHANGE: 'system_purpose_change'
 };
 
-export default Constants;
+const FileConstants = {
+    IMAGE_TYPES: ['jpg', 'gif', 'bmp', 'png', 'jpeg'],
+    AUDIO_TYPES: ['mp3', 'wav', 'wma', 'm4a', 'flac', 'aac', 'ogg'],
+    VIDEO_TYPES: ['mp4', 'avi', 'webm', 'mkv', 'wmv', 'mpg', 'mov', 'flv'],
+    PRESENTATION_TYPES: ['ppt', 'pptx'],
+    SPREADSHEET_TYPES: ['xlsx', 'csv'],
+    WORD_TYPES: ['doc', 'docx'],
+    CODE_TYPES: ['as', 'applescript', 'osascript', 'scpt', 'bash', 'sh', 'zsh', 'clj', 'boot', 'cl2', 'cljc', 'cljs', 'cljs.hl', 'cljscm', 'cljx', 'hic', 'coffee', '_coffee', 'cake', 'cjsx', 'cson', 'iced', 'cpp', 'c', 'cc', 'h', 'c++', 'h++', 'hpp', 'cs', 'csharp', 'css', 'd', 'di', 'dart', 'delphi', 'dpr', 'dfm', 'pas', 'pascal', 'freepascal', 'lazarus', 'lpr', 'lfm', 'diff', 'django', 'jinja', 'dockerfile', 'docker', 'erl', 'f90', 'f95', 'fsharp', 'fs', 'gcode', 'nc', 'go', 'groovy', 'handlebars', 'hbs', 'html.hbs', 'html.handlebars', 'hs', 'hx', 'java', 'jsp', 'js', 'jsx', 'json', 'jl', 'kt', 'ktm', 'kts', 'less', 'lisp', 'lua', 'mk', 'mak', 'md', 'mkdown', 'mkd', 'matlab', 'm', 'mm', 'objc', 'obj-c', 'ml', 'perl', 'pl', 'php', 'php3', 'php4', 'php5', 'php6', 'ps', 'ps1', 'pp', 'py', 'gyp', 'r', 'ruby', 'rb', 'gemspec', 'podspec', 'thor', 'irb', 'rs', 'scala', 'scm', 'sld', 'scss', 'st', 'sql', 'swift', 'tex', 'txt', 'vbnet', 'vb', 'bas', 'vbs', 'v', 'veo', 'xml', 'html', 'xhtml', 'rss', 'atom', 'xsl', 'plist', 'yaml'],
+    PDF_TYPES: ['pdf'],
+    PATCH_TYPES: ['patch'],
+
+    ICON_PATH_FROM_FILE_TYPE: {
```

Comment:
Cool, thanks. That should also make it really easy to use for those other components, especially since we don't have to worry about using those icons in two different ways like the webapp (since some places include them as `<img>` tags and others use CSS).

And yeah, those other constants are fine here. They're just string constants, and I was more concerned with including some string constants also bringing along a bunch of images.